tests/pixel_utils_test.mjs

```diff
@@ -0,0 +1,337 @@
+        const searchExperiments = {
+            no_vars_exp: {
+                description: 'An experiment without variants',
+            },
+        };
+
+        const result = parseSearchExperiments(searchExperiments);
+
+        expect(result.no_vars_exp).to.not.have.property('enum');
+        expect(result.no_vars_exp).to.not.have.property('type');
+        expect(result.prebounce_no_vars_exp).to.not.have.property('enum');
+        expect(result.prebounce_no_vars_exp).to.not.have.property('type');
+    });
+
+    it('should handle multiple experiments', () => {
+        const searchExperiments = {
+            exp1: {
+                description: 'First experiment',
+                variants: ['a', 'b'],
+            },
+            exp2: {
+                description: 'Second experiment',
+                variants: [10, 20],
+            },
+        };
+
+        const result = parseSearchExperiments(searchExperiments);
+
+        expect(result).to.have.all.keys('exp1', 'prebounce_exp1', 'exp2', 'prebounce_exp2');
+
+        expect(result.exp1.type).to.equal('string');
+        expect(result.exp1.description).to.equal('First experiment');
+
+        expect(result.prebounce_exp2.type).to.equal('number');
+        expect(result.prebounce_exp2.key).to.equal('prebounce_exp2');
+        expect(result.prebounce_exp2.description).to.equal('Second experiment');
+    });
+
+    it('should handle experiment definitions with missing properties', () => {
+        const searchExperiments = {
+            valid_exp: {
+                description: 'A valid experiment',
+                variants: ['on', 'off'],
+            },
+            valid_exp_empty: {},
+        };
+
+        expect(() => parseSearchExperiments(searchExperiments)).to.not.throw();
+    });
+
+    it('should throw on experiment definitions that are null or undefined', () => {
+        const searchExperiments = {
+            null_exp: null,
+            undef_exp: undefined,
+        };
+
+        expect(() => parseSearchExperiments(searchExperiments)).to.throw(TypeError);
+    });
+});
+
+describe('matchPixel', () => {
+    const testPixels = {
+        m: {
+            // ...
+        },
+        m_l: {
+            // ...
+        },
+        m_lp: {
+            // ...
+        },
+        m_lp_c: {
+            // ...
+        },
+    };
+
+    const allPixels = {};
+    tokenizePixelDefs(testPixels, allPixels);
+
+    const expectedTokens = {
+        m: allPixels.m[ROOT_PREFIX],
+        m_l: allPixels.m.l[ROOT_PREFIX],
+        m_lp: allPixels.m.lp[ROOT_PREFIX],
+        m_lp_c: allPixels.m.lp.c[ROOT_PREFIX],
+        none: undefined,
+    };
+
+    it('should return the longest matching prefix and the matched pixel object', () => {
+        const [prefix, pixelMatch] = matchPixel('m_lp_a_b', allPixels);
+        expect(prefix).to.equal('m_lp');
+        expect(pixelMatch).to.deep.equal(expectedTokens.m_lp);
```

Comment:
Very minor: but is this check needed since all the defs are identical in this case? Or might be worth to change them up a bit.